Dockerfile.devspace

```diff
@@ -0,0 +1,22 @@
+# Build the manager binary
+FROM --platform=$TARGETPLATFORM golang:alpine3.18 as builder
```

Comment:
I recommend using a specific Go version here that matches the version of Go we're using it prod. So we'd look at `Dockerfile`.

This repo's prod Dockerfile doesn't actually specify a Go version yet. So we should add that too so that prod doesn't end up running old Go versions with CVEs in situations where we haven't done a prod build in a  while. And then make the dev Dockerfile match.

I recommend `FROM --platform=$TARGETPLATFORM golang:1.22-alpine3.19 AS builder` right now.

Reply:
I matched what exists in all of our `Dockerfile.devspace` files for other repos. if you think its a must change, we can change them for all of them, but i dont see a reason to make this specific repo a snowflake.

Reply:
Agreed with Matt on this one. The other devspace configs just need updating.